### PR TITLE
Adding tests for Kubelet pod update functions

### LINKED ISSUE #91313: Add unit test coverage for `pkg/kubelet/types/pod_update.go`
**What would you like to be added**:
Currently, in `pkg/kubelet/types/pod_update.go` the following public methods do not have unit test coverage:
- `IsMirrorPod`
- `IsStaticPod`
- `IsCriticalPod`
- `Preemptable`

These methods are great candidates for adding unit test coverage.

**Why is this needed**:
Adding unit test coverage will both ensure they are working correctly and (more importantly), encode their expected behavior in the test suite. Additionally, if we want to change their behavior in the future, it'll be helpful to already have the test suites.

**Extra note**
I think this could be a great candidate for someone who is new to contributing to sig-node/k8s. I'm happy to mentor any first time contributors :) Please respond saying you'd like to work on this issue and I'll assign it to you. Please reach out to me on k8s slack (where I'm also @mattjmcnaughton) if I can be of any assistance.

### PR DESCRIPTION
**Pull request by knabben**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Increase *k8s.io/kubernetes/pkg/kubelet/types/pod_update.go* coverage from 52.9% to 100.0%

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/91313

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mattjmcnaughton** [comment]
Thanks @knabben - I'll take a look this weekend!

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

Awesome, thank you for expanding the test coverage!

**Comment by prasadkatti** [comment]
@mattjmcnaughton - Are there any other modules that require unit tests that a beginner can work towards? I have added tests for kubeadm. So I a little familiar with the testing process.

File: pkg/kubelet/types/pod_update_test.go [hunk 1]
```diff
@@ -114,6 +148,137 @@ func TestString(t *testing.T) {
...
+func TestPreemptable(t *testing.T) {
+	cases := []struct {
+		preemptor *v1.Pod
+		preemptee *v1.Pod
+		expected  bool
+	}{
+		{
+			preemptor: getTestPod(configSourceAnnotation(FileSource), nil),
+			preemptee: getTestPod(nil, nil),
+			expected:  true,
```
Comment (MHBauer) [inline pkg/kubelet/types/pod_update_test.go:247]: filesource is static pod so it preempts?

File: pkg/kubelet/types/pod_update_test.go [hunk 2]
```diff
@@ -114,6 +148,137 @@ func TestString(t *testing.T) {
...
+		},
+		{
+			preemptor: getTestPod(configSourceAnnotation(ApiserverSource), &systemPriorityUpper),
+			preemptee: getTestPod(configSourceAnnotation(FileSource), &systemPriority),
+			expected:  true,
+		},
+		{
+			preemptor: getTestPod(configSourceAnnotation(FileSource), &systemPriority),
+			preemptee: getTestPod(configSourceAnnotation(FileSource), &systemPriorityUpper),
+			expected:  false,
```
Comment (MHBauer) [inline pkg/kubelet/types/pod_update_test.go:262]: priority, priority, reverse priority, makes sense.

File: pkg/kubelet/types/pod_update_test.go [hunk 3]
```diff
@@ -114,6 +148,137 @@ func TestString(t *testing.T) {
...
+		},
+		{
+			preemptor: getTestPod(configSourceAnnotation(FileSource), nil),
+			preemptee: getTestPod(configSourceAnnotation(FileSource), nil),
+			expected:  false,
+		},
+		{
+			preemptor: getTestPod(configSourceAnnotation(FileSource), &systemPriority),
+			preemptee: getTestPod(configSourceAnnotation(FileSource), &systemPriority),
+			expected:  false,
```
Comment (MHBauer) [inline pkg/kubelet/types/pod_update_test.go:272]: same priority, no preempt. makes sense.

File: pkg/kubelet/types/pod_update_test.go [hunk 4]
```diff
@@ -114,6 +148,137 @@ func TestString(t *testing.T) {
...
+		pod      *v1.Pod
+		expected bool
+	}{
+		{
+			pod:      getTestPod(configSourceAnnotation(FileSource), nil),
+			expected: true,
+		},
+		{
+			pod:      getTestPod(configMirrorAnnotation(), nil),
+			expected: true,
```
Comment (MHBauer) [inline pkg/kubelet/types/pod_update_test.go:212]: file pods and mirror pods are static and thus critical

File: pkg/kubelet/types/pod_update_test.go [hunk 5]
```diff
@@ -114,6 +148,137 @@ func TestString(t *testing.T) {
...
+			pod:      getTestPod(configSourceAnnotation(ApiserverSource), nil),
+			expected: false,
+		},
+		{
+			pod:      getTestPod(nil, &systemPriority),
+			expected: true,
+		},
+		{
+			pod:      getTestPod(nil, &systemPriorityUpper),
+			expected: true,
```
Comment (MHBauer) [inline pkg/kubelet/types/pod_update_test.go:224]: high priority pods are critical, cool.

File: pkg/kubelet/types/pod_update_test.go [hunk 6]
```diff
@@ -114,6 +148,137 @@ func TestString(t *testing.T) {
...
+		},
+		{
+			pod:      getTestPod(configSourceAnnotation(ApiserverSource), nil),
+			expected: false,
+		},
+		{
+			pod:      getTestPod(nil, nil),
+			expected: false,
+		},
```
Comment (MHBauer) [inline pkg/kubelet/types/pod_update_test.go:192]: Is there any difference between ApiserverSource and not? I do not think we need all of the testcases.

**Review by MHBauer — COMMENTED** [review]
It would be nice to have all the cases named and using subtests https://blog.golang.org/subtests so I can run one and see the coverage to understand which path each hits, but otherwise only nit.

File: pkg/kubelet/types/pod_update_test.go [hunk 7]
```diff
@@ -17,6 +17,7 @@ limitations under the License.
 package types
 
 import (
+	"k8s.io/kubernetes/pkg/apis/scheduling"
```
Comment (lsytj0413) [inline pkg/kubelet/types/pod_update_test.go:20]: move this import down to standard library？seems gofmt will regroup the import sentence.

**Comment by knabben** [comment]
@MHBauer PTAL, Added the test cases naming and use the t.Run on them.

**Review by MHBauer — APPROVED** [review]
I like it, I think it is easier to read.
/lgtm

For example
```
--- PASS: TestGetValidatedSources (0.00s)
    --- PASS: TestGetValidatedSources/empty_source (0.00s)
    --- PASS: TestGetValidatedSources/file_and_apiserver_source (0.00s)
    --- PASS: TestGetValidatedSources/all_source (0.00s)
    --- PASS: TestGetValidatedSources/unknown_source (0.00s)
=== RUN   TestGetPodSource
=== RUN   TestGetPodSource/cannot_get_pod_source
=== RUN   TestGetPodSource/valid_annotation_returns_the_source
--- PASS: TestGetPodSource (0.00s)
    --- PASS: TestGetPodSource/cannot_get_pod_source (0.00s)
    --- PASS: TestGetPodSource/valid_annotation_returns_the_source (0.00s)
=== RUN   TestString
=== RUN   TestString/create
=== RUN   TestString/update
=== RUN   TestString/sync
=== RUN   TestString/kill
=== RUN   TestString/unknown
--- PASS: TestString (0.00s)
    --- PASS: TestString/create (0.00s)
    --- PASS: TestString/update (0.00s)
    --- PASS: TestString/sync (0.00s)
    --- PASS: TestString/kill (0.00s)
    --- PASS: TestString/unknown (0.00s)
=== RUN   TestIsMirrorPod
=== RUN   TestIsMirrorPod/mirror_pod
=== RUN   TestIsMirrorPod/not_a_mirror_pod
--- PASS: TestIsMirrorPod (0.00s)
    --- PASS: TestIsMirrorPod/mirror_pod (0.00s)
    --- PASS: TestIsMirrorPod/not_a_mirror_pod (0.00s)
=== RUN   TestIsStaticPod
=== RUN   TestIsStaticPod/static_pod_with_file_source
=== RUN   TestIsStaticPod/static_pod_with_http_source
=== RUN   TestIsStaticPod/static_pod_with_api_server_source
--- PASS: TestIsStaticPod (0.00s)
    --- PASS: TestIsStaticPod/static_pod_with_file_source (0.00s)
    --- PASS: TestIsStaticPod/static_pod_with_http_source (0.00s)
    --- PASS: TestIsStaticPod/static_pod_with_api_server_source (0.00s)
=== RUN   TestIsCriticalPod
=== RUN   TestIsCriticalPod/critical_pod_with_file_source
=== RUN   TestIsCriticalPod/critical_pod_with_mirror_annotation
=== RUN   TestIsCriticalPod/critical_pod_using_system_priority
=== RUN   TestIsCriticalPod/critical_pod_using_greater_than_system_priority
=== RUN   TestIsCriticalPod/not_a_critical_pod_with_api_server_annotation
=== RUN   TestIsCriticalPod/not_critical_if_not_static,_mirror_or_without_a_priority
--- PASS: TestIsCriticalPod (0.00s)
    --- PASS: TestIsCriticalPod/critical_pod_with_file_source (0.00s)
    --- PASS: TestIsCriticalPod/critical_pod_with_mirror_annotation (0.00s)
    --- PASS: TestIsCriticalPod/critical_pod_using_system_priority (0.00s)
    --- PASS: TestIsCriticalPod/critical_pod_using_greater_than_system_priority (0.00s)
    --- PASS: TestIsCriticalPod/not_a_critical_pod_with_api_server_annotation (0.00s)
    --- PASS: TestIsCriticalPod/not_critical_if_not_static,_mirror_or_without_a_priority (0.00s)
=== RUN   TestPreemptable
=== RUN   TestPreemptable/a_critical_preemptor_pod_preempts_a_non_critical_pod
=== RUN   TestPreemptable/a_preemptor_pod_with_higher_priority_preempts_a_critical_pod
=== RUN   TestPreemptable/a_not_critical_pod_with_higher_priority_preempts_a_critical_pod
=== RUN   TestPreemptable/a_critical_pod_with_less_priority_do_not_preempts_a_critical_pod
=== RUN   TestPreemptable/a_critical_pod_without_priority_do_not_preempts_a_critical_pod_without_priority
=== RUN   TestPreemptable/a_critical_pod_with_priority_do_not_preempts_a_critical_pod_with_the_same_priority
--- PASS: TestPreemptable (0.00s)
    --- PASS: TestPreemptable/a_critical_preemptor_pod_preempts_a_non_critical_pod (0.00s)
    --- PASS: TestPreemptable/a_preemptor_pod_with_higher_priority_preempts_a_critical_pod (0.00s)
    --- PASS: TestPreemptable/a_not_critical_pod_with_higher_priority_preempts_a_critical_pod (0.00s)
    --- PASS: TestPreemptable/a_critical_pod_with_less_priority_do_not_preempts_a_critical_pod (0.00s)
    --- PASS: TestPreemptable/a_critical_pod_without_priority_do_not_preempts_a_critical_pod_without_priority (0.00s)
    --- PASS: TestPreemptable/a_critical_pod_with_priority_do_not_preempts_a_critical_pod_with_the_same_priority (0.00s)
=== RUN   TestIsCriticalPodBasedOnPriority
=== RUN   TestIsCriticalPodBasedOnPriority/a_system_critical_pod
=== RUN   TestIsCriticalPodBasedOnPriority/a_non_system_critical_pod
--- PASS: TestIsCriticalPodBasedOnPriority (0.00s)
    --- PASS: TestIsCriticalPodBasedOnPriority/a_system_critical_pod (0.00s)
    --- PASS: TestIsCriticalPodBasedOnPriority/a_non_system_critical_pod (0.00s)
```

**Comment by knabben** [comment]
ping @derekwaynecarr

**Pull request merged**